gas/config/tc-riscv.c

```diff
@@ -1083,6 +1118,7 @@ validate_riscv_insn (const struct riscv_opcode *opc, int length)
       case 'P': USE_BITS (OP_MASK_PRED, OP_SH_PRED); break;
       case 'Q': USE_BITS (OP_MASK_SUCC, OP_SH_SUCC); break;
       case 'o': /* ITYPE immediate, load displacement.  */
+      case 'l': /* IMM6L */
```

Comment:
This looks strange.  We encode it as ENCODE_ITYPE_IMM here (validate_riscv_insn), but will use ENCODE_SBTYPE_IMM6L to encode it later in the riscv_ip.

Reply:
Thanks for pointing it out. Fixed it the latest commit.